### ISSUE TITLE
Error calling Set-PowerlinePrompt after update to 2.3.1

### ISSUE THREAD

**Issue by weloytty** [Jaykul/PowerLine#16]
So I updated to 2.3.1, and now my profile won't load.  I stripped it down To the bare minimum, and we get


using module PowerLine
using namespace PowerLine

Set-StrictMode -Version Latest
Set-PSDebug -Strict -Trace 0
$PowerLinePrompt = @(
        @{ bg = "Cyan";     fg = "White"; text = { $MyInvocation.HistoryId } }
        @{ bg = "DarkBlue"; fg = "White"; text = { $pwd } }
    )

if ($Env:ConEmuANSI -or $Host.UI.SupportsVirtualTerminal) {
  Set-PowerLinePrompt -CurrentDirectory -PowerLineFont -Title { "PowerShell - {0} ({1})" -f (Convert-Path $pwd),$pwd.Provider.Name }
} 


and I recieve the error:
> The property 'Title' cannot be found on this object. Verify that the property exists and can be set.
> At C:\Program Files\WindowsPowerShell\Modules\PowerLine\2.3.1\PowerLine.psm1:414 char:9
> +         $global:PowerLinePrompt.Title = $Title
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : PropertyAssignmentException`
> 
> The property 'SetCurrentDirectory' cannot be found on this object. Verify that the property exists and can be set.
> At C:\Program Files\WindowsPowerShell\Modules\PowerLine\2.3.1\PowerLine.psm1:417 char:9
> +         $global:PowerLinePrompt.SetCurrentDirectory = $CurrentDirecto ...
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : PropertyAssignmentException`

> The property 'UseAnsiEscapes' cannot be found on this object. Verify that the property exists and can be set.
> At C:\Program Files\WindowsPowerShell\Modules\PowerLine\2.3.1\PowerLine.psm1:420 char:5
> +     $global:PowerLinePrompt.UseAnsiEscapes = $UseAnsiEscapes
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : PropertyAssignmentException`

> The property 'RestoreVirtualTerminal' cannot be found on this object. Verify that the property exists and can be set. At C:\Program Files\WindowsPowerShell\Modules\PowerLine\2.3.1\PowerLine.psm1:421 char:5
> +     $global:PowerLinePrompt.RestoreVirtualTerminal = $RestoreVirtualT ...
> +     
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : PropertyAssignmentException`



PS>get-module pansies|select Name, Version, ModuleBase                                      
                                                                                            
Name    Version ModuleBase                                                                  
----    ------- ----------                                                                  
Pansies 1.0.0   C:\Program Files\WindowsPowerShell\Modules\Pansies\1.0.0    `                
                                                                                            
                                                                                            
PS>get-module powerline|select Name, Version, ModuleBase                                    
                                                                                            
Name      Version ModuleBase                                                                
----      ------- ----------                                                                
PowerLine 2.3.1   C:\Program Files\WindowsPowerShell\Modules\PowerLine\2.3.1    `            



PSVersionTable:

PS>$PSVersionTable                                      
                                                        
Name                           Value                    
----                           -----                    
PSVersion                      5.1.14409.1005           
PSEdition                      Desktop                  
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}  
BuildVersion                   10.0.14409.1005          
CLRVersion                     4.0.30319.42000          
WSManStackVersion              3.0                      
PSRemotingProtocolVersion      2.3                      
SerializationVersion           1.1.0.1                  
                                                        

Running on Windows7 sp1 x64.

I think I'm missing something simple.

**Comment by weloytty** [comment]
so if I explicitly identify $PowerLinePrompt as [PowerLine.Prompt], I get a bit further, however, the custom prompt:

`[PowerLine.Prompt]$PowerLinePrompt = 1,
 @(
    $null,
    @(
      @{ text = { New-PowerLineBlock (Get-Elapsed) -ErrorBackgroundColor DarkRed -ErrorForegroundColor White -ForegroundColor Black -BackgroundColor DarkGray } }
      @{ bg = "Gray"; fg = "Black"; text = { Get-Date -f "T" } }
    )
  ),
   @(
    @{ bg = "Red"; fg = "Yellow"; text = {
        if ($Global:IsAdmin) {
          "ADMIN"
        } else {
          ""
        }
      }
    },
    @{ bg = "Yellow"; fg = "Red"; text = { $Env:ComputerName } },
    @{ bg = "DarkBlue"; fg = "White"; text = { 
        $location = $(Get-Location).Path
        #$location = $location.Replace("Microsoft.PowerShell.Core\FileSystem::","")
        Convert-Path -Path $location
      } 
    }
  )`

Gets me a prompt of 

PS>


if I replace the $null in the first line with "", the prompt will display something -- just the second line though (first line is still omitted)

**Comment by weloytty** [comment]
$So then I started looking at $error.

I resolved the problem by specifiying the [RgbColor] Parameters as [PoshCode.Pansies.RgbColor]

I have no idea on the crossplatform implications of this, it only served to get it running on my work machine, on win7sp1.  I'll try it on my home win10 machine as soon as I get there.

**Comment by dotps1** [comment]
i am also getting the last two errors from the orginal post,
> The property 'UseAnsiEscapes' cannot be found on this object. Verify that the property exists and can be set.
> The property 'RestoreVirtualTerminal' cannot be found on this object. Verify that the property exists and can be set.

Running Win10 x64 1702

**Comment by mweitzel2005** [comment]
Am getting the same on Win10 x64 1703.

**Comment by weloytty** [comment]
I just submitted pull request #20  to address

**Comment by uberrich** [comment]
I had the same problem as @dotps1, but as suggested by @weloytty I changed `$PowerLinePrompt` to `[PowerLine.Prompt]$PowerLinePrompt`, and also manually changed my powerline.psm1 file to add the changes from #20 and it worked! Yay! :-)

**Comment by Jaykul** [comment]
I don't understand how these issues escaped my notice. This was a busy summer, but I'm really frustrated about this. I thought I'd been monitoring things well! At least since August ;-)

Anyway. I ripped this all apart and putt it back together much more simply in 3.0 -- which I'm ready to release this weekend, so I'm not going to try and address this any further. I pulled the changes in #20 even though I don't understand why you needed them, so I'm going to suggest you try 3.0 and let me know if it doesn't address this.

The configurations are going to be **much** simpler, which unfortunately is a breaking change -- but I think when you try to use it, you'll agree it's better 😉

**Comment by weloytty** [comment]
Once I figured out how 3.0 worked, I agree -- I do like it better, and I have it working for me.  And dont worry about not seeing the issue, it looks as if you have a lot on your plate!

I fear though, that the problem still exists, and I can't figure out why:  I have to change RgbColor to PoshCode.Pansies.RbgColor but I also have to change [List] to [System.Collections.Generic.List].  It's got to be some environmental thing, but I get the behavior on both my work Win10 (Anniversary) laptop and my home machine (Win10 whatever the slow ring is currently running).

I'll submit a pull request here in a few to show what I did to make it work again for me.
